util/utils.cpp

```diff
@@ -6,6 +6,12 @@
 #include <gflags/gflags.h>
 #include <numeric>
 
+DEFINE_string(dist,
+              "l2",
+              "Algorithm to use for distance computation of sketch methods. "
+              "l1|l2|exp");
```

Comment:
I'd like to avoid adding flags to libraries, if possible. Sometimes it's really useful to have a flag, but in this case, I think adding a flag can be easily avoided by simply moving the `sketch_dist` function (and the `dist` flag) to the caller.

Reply:
Yeah I kinda agree. It's just that this otherwise simple change would have impact on all the Sketch constructors (making them take and pass on the flag), which is a bit annoying.

I could refactor things so that the constructors take a struct with options instead, which might be cleaner at this point and make adding more options easier.